Classes/DataProcessing/ImageProcessor.php

```diff
@@ -0,0 +1,233 @@
+            $processedData[$targetFieldName] = $this->processFiles();
+        }
+
+        return $processedData;
+    }
+
+    /**
+     * @return array
+     */
+    protected function processFiles(): array
+    {
+        $data = [];
+
+        foreach ($this->fileObjects as $fileObject) {
+            $metaData = $fileObject->toArray();
+            $data['images'][] = [
+                'publicUrl' => $this->getImageService()->getImageUri($fileObject, true),
+                'dimensions' => [
+                    'width' => $fileObject->getProperty('width'),
+                    'height' => $fileObject->getProperty('height')
+                ],
+                'metaData' => [
+                    'title' => $metaData['title'],
+                    'alternative' => $metaData['alternative'],
+                    'description' => $metaData['description'],
+                    'link' => !empty($metaData['link']) ? $this->contentObjectRenderer->typoLink_URL([
+                        'parameter' => $metaData['link']
+                    ]) : null
+                ]
+            ];
+        }
+
+        return $data;
+    }
+
+    /**
+     * @return array
+     */
+    protected function processGalleryFiles(): array
+    {
+        $data = [];
+
+        foreach ($this->galleryObjects['rows'] as $rowKey => $row) {
+            foreach ($row['columns'] as $columnKey => $image) {
+                /**
+                 * @var $processedFile ProcessedFile
+                 */
+                $processedFile = $this->processImageFile($image['media'], [
+                    'width' => $image['dimensions']['width'],
+                    'height' => $image['dimensions']['height']
+                ]);
+                $metaData = $image['media']->toArray();
+                $data['images'][] = [
+                    'publicUrl' => $this->getImageService()->getImageUri($processedFile, true),
+                    'dimensions' => [
+                        'width' => $processedFile->getProperty('width'),
+                        'height' => $processedFile->getProperty('height')
+                    ],
+                    'metaData' => [
+                        'title' => $metaData['title'],
+                        'alternative' => $metaData['alternative'],
+                        'description' => $metaData['description'],
+                        'link' => !empty($metaData['link']) ? $this->contentObjectRenderer->typoLink_URL([
+                            'parameter' => $metaData['link']
+                            ]) : null
+                    ]
+                ];
+            }
+        }
+        $data['gallery'] = $this->galleryObjects;
+
+        return $data;
+    }
+
+    /**
+     * @param FileReference $image
+     * @param array $dimensions
+     * @return ProcessedFile
+     */
+    protected function processImageFile(FileReference $image, array $dimensions): ProcessedFile
```

Comment:
is this code taken somewhere from TYPO3 core?

Reply:
yes, from ImageViewHelper